ExomeDepth/igv_xml_session.py

```diff
@@ -9,34 +9,39 @@
 
 def parse_ped(ped_file):
     samples = {}  # 'sample_id': {'family': 'fam_id', 'parents': ['sample_id', 'sample_id']}
-    for line in ped_file:
-        ped_data = line.strip().split()
-        family, sample, father, mother, sex, phenotype = ped_data
-
-        # Create samples
-        if sample not in samples:
-            samples[sample] = {'family': family, 'parents': [], 'children': []}
-        if father != '0' and father not in samples:
-            samples[father] = {'family': family, 'parents': [], 'children': []}
-        if mother != '0' and mother not in samples:
-            samples[mother] = {'family': family, 'parents': [], 'children': []}
-
-        # Save sample relations
-        if father != '0':
-            samples[sample]['parents'].append(father)
-            samples[father]['children'].append(sample)
-        if mother != '0':
-            samples[sample]['parents'].append(mother)
-            samples[mother]['children'].append(sample)
+    with open(ped_file, 'r') as ped_file_lines:
+        for line in ped_file_lines:
+            ped_data = line.strip().split()
+            family, sample, father, mother, sex, phenotype = ped_data
+
+            # Create samples
+            if sample not in samples:
+                samples[sample] = {'family': family, 'parents': [], 'children': []}
+            if father != '0' and father not in samples:
+                samples[father] = {'family': family, 'parents': [], 'children': []}
+            if mother != '0' and mother not in samples:
+                samples[mother] = {'family': family, 'parents': [], 'children': []}
+
+            # Save sample relations
+            if father != '0':
+                samples[sample]['parents'].append(father)
+                samples[father]['children'].append(sample)
+            if mother != '0':
+                samples[sample]['parents'].append(mother)
+                samples[mother]['children'].append(sample)
     return samples
 
 
-def make_reanalysis(reanalysis_file):
+def parse_reanalysis_file(reanalysis_file):
     reanalysis = {}
     for line in reanalysis_file:
-        splitline = line.split()
-        if splitline[0] not in reanalysis:
-            reanalysis[splitline[0]] = {"refset": splitline[1], "suffix": splitline[2]}
+        sampleid, refset, tag = (
+            lambda sampleid, refset, tag = None:(sampleid, refset, tag)
+        )(
+            *line.strip().split()
+        )
+
+        reanalysis[sampleid] = {"refset": refset, "tag": tag}
```

Comment:
This looks very complicated for an easy task? What are you trying to achieve?

Reply:
The difficulty here is that column 3 (tag) is allowed to be empty.
Could not come up with an easy solution, so found it 'elsewhere'.... if you have a better one, let me know!